s3torchconnector/src/s3torchconnector/lightning/s3_lightning_checkpoint.py

```diff
@@ -0,0 +1,84 @@
+#  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+#  // SPDX-License-Identifier: BSD
+
+from typing import Optional, Dict, Any
+
+import torch
+
+from lightning.pytorch.plugins.io import CheckpointIO
+
+from .._s3client import S3Client
+from .._s3dataset_common import parse_s3_uri
+
+
+class S3LightningCheckpoint(CheckpointIO):
+    """A checkpoint manager for S3 using the :class:`CheckpointIO` interface."""
+
+    def __init__(self, region: str):
+        self.region = region
+        self._client = S3Client(region)
+
+    def save_checkpoint(
+        self,
+        checkpoint: Dict[str, Any],
+        path: str,
+        storage_options: Optional[Any] = None,
+    ) -> None:
+        """Save model/training states as a checkpoint file through state-dump and upload to S3.
+
+        Args:
+            checkpoint: dict containing model and trainer state
```

Comment:
nit: consistency

> checkpoint: dict
Should be checkpoint: Dict ... 
Same for path description.